### PR TITLE
feat(cat-voices): provide seed phrase 953

### LINKED ISSUE #953: Provide SeedPhrase

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

- Reusable `VoicesAutocomplete` + `SeedPhraseField`
- `SeedPhraseInputPanel` from recover flow

## Breaking Changes

Describe any breaking changes and the impact.

## Demo

https://github.com/user-attachments/assets/106e2da5-8be8-48a6-9f7f-922de97470c6

## Related Pull Requests

Closes #953 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Review by digitalheartxs — DISMISSED** [review]

File: catalyst_voices/lib/pages/registration/recover/seed_phrase/seed_phrase_input_panel.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    // TODO(damian-molinski): Import implementation for KeychainCreation
+  }
+
+  void _resetControllerWords() {
+    _controller.clear();
+    _focusNode.requestFocus();
+  }
+
+  void _onWordsChanged() {
+    final words = [..._controller.value];
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/registration/recover/seed_phrase/seed_phrase_input_panel.dart:77]: If your intention was to create a copy of a list then List.of() seems a better alternative then using a spread operator

File: catalyst_voices/lib/widgets/text_field/voices_autocomplete.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        padding: const EdgeInsets.only(top: 6),
+        child: ConstrainedBox(
+          constraints: const BoxConstraints(maxWidth: 300),
+          child: Material(
+            color: Theme.of(context).colors.elevationsOnSurfaceNeutralLv1White,
+            elevation: 2,
+            borderRadius: BorderRadius.circular(4),
+            // Clip is here for InkWell border radius.
+            child: ClipRRect(
+              borderRadius: BorderRadius.circular(4),
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/text_field/voices_autocomplete.dart:90]: Material widget which is above has ClipBehavior as well, you should be able to set it to Clip.antiAlias, etc. Though I'm not sure if that works with every MaterialType

Comment (damian-molinski) [reply catalyst_voices/lib/widgets/text_field/voices_autocomplete.dart:90]: Actually `clipBehavior: Clip.antiAlias`, love it!

**Review by dtscalac — APPROVED** [review]
LGTM

**Pull request merged**